### PR TITLE
CQ-4347957 Prefill Support

### PR DESCRIPTION
**Pull request by nit23uec**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/utils.js [hunk 1]
```diff
@@ -140,10 +124,22 @@ export default class Utils {
             if (_path == null) {
                 console.error(`data-${Constants.NS}-${formContainerClass}-path attribute is not present in the HTML element. Form cannot be initialized` )
             } else {
-                const _formJson = await Utils.getJson(_path + ".model.json");
-                console.log("model json from server ", _formJson);
+                const _formsList = await HTTPAPILayer.getFormsList();
```
[the remark this answers is not in the document]
Comment (nit23uec) [reply ui.frontend/src/utils.js:127]: we should not expose the id generation logic in public.

File: ui.frontend/src/view/FormContainer.js [hunk 2]
```diff
@@ -20,6 +20,9 @@ export default class FormContainer {
         this._path = params._path;
         this._fields = {};
         this._deferredParents = {};
+        if (params._prefillData) {
+            this._model.importData(params._prefillData.data);
+        }
```
Comment (vdua) [inline ui.frontend/src/view/FormContainer.js:25]: I think better would be 
`createFormInstance({...params._formJson, data: _params._prefillData})`

**Comment by vdua** [comment]
Since it requires latest aemds (af2-rest-api), moving it to draft state.

File: ui.frontend/src/HTTPAPILayer.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        } else {
+            //TODO: throw errors once API is available on Circle CI set up
+            console.debug("Error in fetching form");
+        }
+    }
+
+    static async getFormDefinition(formContainerPath) {
+        return await this.#getJson(formContainerPath + ".model.json");
+    }
+
```
Comment (vdua) [inline ui.frontend/src/HTTPAPILayer.js:34]: I am still not comfortable with making so many network calls to get the form definition.

Comment (nit23uec) [reply ui.frontend/src/HTTPAPILayer.js:34]: there's still only one call required to fetch the model definition.
but for prefill call, we need form id, and hence, we will need listForms call.
issue is already logged to optimise it. 
https://github.com/adobe/aem-core-forms-components/issues/190

**Review by vdua — APPROVED** [review]

**Pull request merged**